### ISSUE TITLE
Migrate from JUnit4 to JUnit5

### ISSUE THREAD

**Issue by Captain1653** [googleapis/google-auth-library-java#756]
The library uses JUnit4 . JUnit5 is stable. Is it worth migrating? I can do it. It's interesting for me.

It can be something like that:
- update maven-surfire-plugin (if necessary)
- update junit version
- run old tests with vintage junit runner
- refactor old tests step by step

Kind regards

**Comment by suztomo** [comment]
HI @Captain1653 . Thank you for filling the issue. Just curious. Is this (this repository's using JUnit4) blocking some of your activity?

**Comment by Captain1653** [comment]
No, it doesn't block my activity. It's only my opinion and I can do it. Does it make sense to start ?

**Comment by Capstan** [comment]
This renders this package incapable of being imported into Google's own monorepo due to trailing support for JUnit5 in bazel Java.
